### DISCUSSION TITLE
Cursor column position not preserved when navigating between lines.

### DISCUSSION THREAD

**Discussion by zjr** [JetBrains/ideavim#471] [Q&A]
When I move the cursor from line to line, I generally expect Vim to preserve the column position. Recently I find the column position is lost when traveling over shorter lines, to the last character in the shortest line. 

I'm trying to illustrate here…

```
_________[1]___
___[2]
___[3]___[x]_____
```

1. starting cursor position
2. shorter line, cursor moves to end of line
3. cursor now at end of line column for previous line, where I would have expected it to be at position `x`.

Is this a setting somewhere?

**Comment by AlexPl292** [comment]
Well, this actually should work well. Could you please attach .ideavimrc file and a small screencast?

**Reply by zjr** [reply]
I was making a scratch file for the video and I found that it actually works as expected in my simple scratch file, but for whatever reason starts to break in a more complicated context.

Here's my .ideavimrc:

```bash
set clipboard+=unnamed

set surround
nnoremap <Leader><Key> T>dt<

set visualbell
set noerrorbells

set dialogescape=on

set relativenumber number
```

https://user-images.githubusercontent.com/913633/159299595-8322b163-a249-470f-a942-819d1815eedc.mov

In this video, I'm starting on the `S` in `fbAdsSdk`, and everything works as expected until I get to L14.

**Reply by AlexPl292** [reply]
Hey, after some investigation, it seems like there was an issue in java runtime that calculates a tab character size a little bit inaccurate. So, after some number of tab characters, it results into this issue.
This issue should be fixed with the next release of IntelliJ IDEA.

**Reply by zjr** [reply]
Fantastic, thank you very much!
